apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/index.js

```diff
@@ -123,22 +145,31 @@ class PageTemplateModal extends Component {
 				overlayClassName="page-template-modal-screen-overlay"
 			>
 				<div className="page-template-modal__inner">
-					<form className="page-template-modal__form">
-						<fieldset className="page-template-modal__list">
-							<TemplateSelectorControl
-								label={ __( 'Template', 'full-site-editing' ) }
-								templates={ this.props.templates }
-								blocksByTemplates={ this.state.blocks }
-								onTemplateSelect={ this.previewTemplate }
-								useDynamicPreview={ true }
-								siteInformation={ siteInformation }
+					{ this.state.isLoading ? (
+						<div className="page-template-modal__loading">
+							<Spinner />
+							{ __( 'Inserting template…', 'full-site-editing' ) }
+						</div>
+					) : (
```

Comment:
Anytime I come across this, I wonder if it may be time to separate it out into a function.